src/test/scala/mesosphere/marathon/integration/MesosAppIntegrationTest.scala

```diff
@@ -372,5 +378,52 @@ class MesosAppIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonT
       status2 should be(OK)
       status2.value.instances.filter(_.status == raml.PodInstanceState.Stable) should have size 2
     }
+
+    "deploy a simple pod with unique constraint and then " taggedAs WhenEnvSet(envVar, default = "true") in {
+
+      val constraints = Set(
+        Constraint.newBuilder
+        .setField(hostnameField)
+        .setOperator(UNIQUE)
+        .build
+      )
+
+      Given("a pod with a single task")
+      val podName = "simple-pod-with-unique-constraint"
+      val pod = simplePod(podName, constraints = constraints, instances = 1)
+
+      When("The pod is deployed")
+      val createResult = marathon.createPodV2(pod)
+
+      Then("The pod is created")
+      createResult should be(Created)
+      waitForDeployment(createResult)
+      waitForPod(pod.id)
+
+      When("The pod config is updated")
+      val scaledPod = pod.copy(instances = 2)
+      val updateResult = marathon.updatePod(pod.id, scaledPod, force = true)
+
+      Then("The pod is not scaled")
+      updateResult should be(OK)
+      val queueResult = marathon.launchQueue()
+      val jsQueueResult = queueResult.entityJson
+
+      val queuedRunspecs = (jsQueueResult \ "queue").as[Seq[JsObject]]
+      val jsonPod = queuedRunspecs.find { spec => (spec \ "pod" \ "id").as[String] == s"/$podName" }.get
+
+      val unfulfilledConstraintRejectSummary = (jsonPod \ "processedOffersSummary" \ "rejectSummaryLastOffers").as[Seq[JsObject]]
+        .find { e => (e \ "reason").as[String] == "UnfulfilledConstraint" }.get
+
+      And("unique constraint reject must happen")
+      (unfulfilledConstraintRejectSummary \ "declined").as[Int] should be >= 1
```

Comment:
You might be too fast with this request (if no offer was matched yet). I would put an `eventually` around it.